api/envoy/config/listener/v3/quic_config.proto

```diff
@@ -24,11 +24,25 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // [#protodoc-title: QUIC listener config]
 
 // Configuration specific to the UDP QUIC listener.
-// [#next-free-field: 12]
+// [#next-free-field: 13]
 message QuicProtocolOptions {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.api.v2.listener.QuicProtocolOptions";
 
+  // Configuration for UDP cmsg to pass to listener filters via QuicReceivedPacket.
+  // Both level and type must be specified for cmsg to be saved.
+  // More info about CMSG: https://man7.org/linux/man-pages/man3/cmsg.3.html
+  message SaveCmsgConfig {
+    // cmsg level to save if matched. Default is unset and no cmsg will be saved.
+    google.protobuf.Int32Value level = 1;
```

Comment:
Looking a bit closer at CMSG, it seems that the type for `level`, `type` and `expected_size` should be unsigned.

Reply:
Done; this requires a static cast to int for equality check with the [cmsghdr int types](https://man7.org/linux/man-pages/man3/cmsg.3.html).